packages/service/src/storybook.utils.ts

```diff
@@ -0,0 +1,380 @@
+            const startIndex = shard * storiesPerShard
+            const endIndex = Math.min(startIndex + storiesPerShard, totalStories)
+            const shardStories = storiesArray.slice(startIndex, endIndex)
+            const testContent = createTestCont({ ...createTestContentData, stories: shardStories })
+            const fileId = `${fileNamePrefix}-${shard + 1}-${numShards}`
+            const describeTitle = `Shard ${shard + 1} of ${numShards}`
+            const fileData = createFileD(describeTitle, testContent)
+
+            writeTestF(directoryPath, fileId, log, fileData)
+        }
+    }
+}
+
+/**
+ * Create the storybook capabilities based on the specified browsers
+ */
+export function createStorybookCapabilities(capabilities: Capabilities.RemoteCapabilities, log: Logger) {
+    const isHeadless = getArgvValue('--headless', value => value !== 'false') ?? true
+    const browsers = getArgvValue('--browsers', (value: string) => value.split(',')) ?? ['chrome']
+
+    if (Array.isArray(capabilities)) {
+        const chromeCapability = {
+            browserName: 'chrome',
+            'goog:chromeOptions': {
+                args: [
+                    'disable-infobars',
+                    ...(isHeadless ? ['--headless'] : []),
+                ],
+            },
+            'wdio-ics:options': {
+                logName: 'local-chrome',
+            },
+        }
+        const firefoxCapability = {
+            browserName: 'firefox',
+            'moz:firefoxOptions': {
+                args: [...(isHeadless ? ['-headless'] : []),]
+            },
+            'wdio-ics:options': {
+                logName: 'local-firefox',
+            },
+        }
+        const safariCapability = {
+            browserName: 'safari',
+            'wdio-ics:options': {
+                logName: 'local-safari',
+            },
+        }
+        const edgeCapability = {
+            browserName: 'MicrosoftEdge',
+            'ms:edgeOptions': {
+                args: [...(isHeadless ? ['--headless'] : [])]
+            },
+            'wdio-ics:options': {
+                logName: 'local-edge',
+            },
+        }
+        interface CapabilityMap {
+            chrome: typeof chromeCapability;
+            firefox: typeof firefoxCapability;
+            safari: typeof safariCapability;
+            edge: typeof edgeCapability;
+        }
+        const capabilityMap: CapabilityMap = {
+            chrome: chromeCapability,
+            firefox: firefoxCapability,
+            safari: safariCapability,
+            edge: edgeCapability,
+        }
+        const newCapabilities = browsers
+            .filter((browser:string): browser is keyof CapabilityMap => browser in capabilityMap)
+            .map((browser:string) => capabilityMap[browser as keyof CapabilityMap])
+        capabilities.length = 0
+        // Add the new capability to the capabilities array
+        // @ts-ignore
```

Comment:
Anything we can do to resolve this?

Reply:
It's solved by adding the correct type as you suggested, didn't think about this. Thanks!